packages/gatsby/src/utils/api-node-docs.js

```diff
@@ -87,8 +87,31 @@ exports.createPagesStatefully = true
  * creating nodes.
  *
  * See also the documentation for [`createNode`](/docs/actions/#createNode).
+ *
+ * @param {object} $0
+ * @param {object} $0.actions
+ * @param {string} $0.basePath
+ * @param {string} $0.pathPrefix
+ * @param {object} $0.boundActionCreators
+ * @param {object} $0.actions
+ * @param {function} $0.loadNodeContent
+ * @param {object} $0.store
+ * @param {object} $0.emitter
+ * @param {function} $0.getCache
+ * @param {function} $0.getNodes
+ * @param {function} $0.getNode
+ * @param {function} $0.getNodesByType
+ * @param {boolean} $0.hasNodeChanged
+ * @param {object} $0.reporter
+ * @param {function} $0.getNodeAndSavePathDependency
+ * @param {object} $0.cache
+ * @param {function} $0.createNodeId
+ * @param {function} $0.createContentDigest
+ * @param {object} $0.tracing
+ * @param {object} $0.schema
```

Comment:
We are documenting those in https://www.gatsbyjs.org/docs/node-api-helpers/ but this thing is never referenced there ;/ I think it would be better to reference that than adding those to all of our hooks?

Reply:
I'd rather we add to add since people tend to reference this page when looking for a certain lifecycle and then get confused because they feel like maybe it _doesn't_ receive certain params

Reply:
well - then those should have descriptions etc - just blank list of things doesn't provide much value?

Reply:
Alternatively we could have custom code that merges params from here with those from common things passed as 1st arg?

Reply:
Agreed, should definitely have descriptions 😛 

Will add in a separate PR?

Reply:
Also - if we are doing that you should add all of those to every lifecycle and not just `sourceNodes` - this doesn't seem like something that can be maintainable?

Reply:
You're absolutely right. Considering all apis get most of these, can I somehow generate them for every api from a common jsdoc or something like that? 

Otherwise not quite maintainable indeed

Reply:
Wait, I have a better idea. I'll update `index.d.ts` to ensure that is up to date and then _import_ those in here.